recipes-kernel/linux/files/patches-6.1/0081-iio-light-Add-support-for-everlight-pmd16d17-sensor.patch

```diff
@@ -0,0 +1,408 @@
++
++			ret = pm16d17_read_reg(data, PM16D17_PS_DATA_H, &ps_data_h);
++			if (ret < 0)
++				goto fail;
++
++			ps_data = (ps_data_h << 8) | ps_data_l;
++
++			dev_dbg(&data->client->dev, "PS data: %x\n", ps_data);
++
++			*val = ps_data;
++			ret = IIO_VAL_INT;
++			break;
++		default:
++			break;
++		}
++	default:
++		break;
++	}
++fail:
++	mutex_unlock(&data->lock);
++
++	return ret;
++}
++
++static int pm16d17_write_raw(struct iio_dev *indio_dev,
++			     struct iio_chan_spec const *chan,
++			     int val, int val2, long mask)
++{
++	return 0;
++}
++
++
++static int pm16d17_write_raw_get_fmt(struct iio_dev *indio_dev,
++				     struct iio_chan_spec const *chan,
++				     long mask)
++{
++	return 0;
++}
++
++static const struct iio_info pm16d17_info = {
++	.read_raw = pm16d17_read_raw,
++	.write_raw = pm16d17_write_raw,
++	.write_raw_get_fmt = pm16d17_write_raw_get_fmt,
++};
++
++static int pm16d17_chip_init(struct pm16d17_data *data)
++{
++	struct i2c_client *client = data->client;
++	struct device_node *np = client->dev.of_node;
++	const char *conv_time = NULL;
++	const char *wait_time = NULL;
++	uint8_t op_mode_setting_val;
++	uint32_t ps_offset_cancel;
++	uint8_t offset_lsb;
++	uint8_t offset_msb;
++	uint32_t pulse_count;
++	uint32_t pgain;
++	unsigned int val;
++	int ret;
++
++	ret = pm16d17_read_reg(data, PM16D17_DEV_ID, &val);
++
++	if (ret < 0 || (val != DEVICE_ID)) {
++		dev_err(&client->dev, "Invalid chip id 0x%04x\n", val);
++		return -ENODEV;
++	}
++
++	dev_dbg(&client->dev, "Detected PM16D17 with chip id: 0x%04x\n", val);
++
++	ret = pm16d17_write_reg(data, PM16D17_OP_MODE, ENABLE_PS_FUNCTION);
++	if (ret < 0)
++		return ret;
++
++	of_property_read_u32(np, "ps-gain", &pgain);
++	switch (pgain)
++	{
++	case 1:
++	case 2:
++	case 4:
++	case 8:
++		op_mode_setting_val |= (ilog2(pgain) << 6) & PS_GAIN_MASK;
++		break;
++	default:
++		break;
++	}
++
++	/* TBD: what if ps-itime is not in dtb, this should return null */
++	of_property_read_string(np, "ps-itime", &conv_time);
++	if (strcmp(conv_time, "0.4") == 0)
++		op_mode_setting_val |= PITIME_0_POINT_4_MS & PS_ITIME_MASK;
++	else if (strcmp(conv_time, "0.8") == 0)
++		op_mode_setting_val |= PITIME_0_POINT_8_MS & PS_ITIME_MASK;
++	else if (strcmp(conv_time, "1.6") == 0)
++		op_mode_setting_val |= PITIME_1_POINT_6_MS & PS_ITIME_MASK;
++	else if (strcmp(conv_time, "3.2") == 0)
++		op_mode_setting_val |= PITIME_3_POINT_2_MS & PS_ITIME_MASK;
++	else if (strcmp(conv_time, "6.3") == 0)
++		op_mode_setting_val |= PITIME_6_POINT_3_MS & PS_ITIME_MASK;
++	else if (strcmp(conv_time, "12.6") == 0)
++		op_mode_setting_val |= PITIME_12_POINT_6_MS & PS_ITIME_MASK;
++	else if (strcmp(conv_time, "25.2") == 0)
++		op_mode_setting_val |= PITIME_25_POINT_2_MS & PS_ITIME_MASK;
++	else {
++		dev_info(&client->dev, "Using default ps itime value\n");
++		op_mode_setting_val |= PITIME_0_POINT_4_MS & PS_ITIME_MASK;
++	}
++
++	/* TBD: what if ps-wtime is not in dtb, this should return null */
```

Comment:
See above.

Reply:
Removed, fixed in https://github.com/siemens/meta-iot2050/pull/509